module-apps/application-messages/data/SMSdata.hpp

```diff
@@ -72,18 +72,25 @@ class SMSSendRequest : public SMSRequest
 class SMSSendTemplateRequest : public SMSRequest
 {
   public:
-    explicit SMSSendTemplateRequest(const utils::PhoneNumber::View &phoneNumber, bool preventAutoLock = false)
-        : SMSRequest(phoneNumber), preventAutoLock(preventAutoLock)
+    enum class AutolockBehavior
```

Comment:
I think it would be nice to consider in the future to expand this mechanism everywhere not only in this particular message because in the end preventAutolock is still a bool flag

Reply:
You're right, but it'd require a refactor in several places in code that IMO is out of scope of this task. The change I made here is to avoid passing bool flag as a function parameter, as it greatly reduces readability of the function call. It's hard to determine what `true` means in `foo(someObj, someData, true)` at the first glance, unless the function name specifies it explicitly.
Anyway, that's a good remark indeed and we can consider it for some future refactoring ;)